src/test/java/org/opentripplanner/routing/core/AccessibilityRoutingTest.java

```diff
@@ -0,0 +1,304 @@
+
+        // we are banning the blue line even though it is accessible to force getting on the green one
+        Itinerary i = getTripPlan(start, end, r -> r.setBannedRoutes("MARTA_BLUE")).itinerary.get(0);
+        List<Leg> transitLegs = i.legs.stream().filter(Leg::isTransitLeg).collect(Collectors.toList());
+
+        Leg leg = transitLegs.get(0);
+        assertEquals("GREEN", leg.routeShortName);
+
+        assertEquals("GEORGIA STATE STATION", leg.from.name);
+        assertEquals("ASHBY STATION", leg.to.name);
+
+        // because are using a trip that has unknown accessibility, it has a lower score
+        assertEquals(0.8333f, leg.accessibilityScore, DELTA);
+    }
+
+    @Test
+    public void canApplyStairsPenalty() {
+        GenericLocation start = new GenericLocation(33.75630, -84.39527);
+        GenericLocation end = new GenericLocation(33.75649, -84.39580);
+
+        Itinerary i = getTripPlan(start, end, r -> r.setMode(TraverseMode.WALK)).itinerary.get(0);
+        Leg leg = i.legs.get(0);
+        assertEquals("WALK", leg.mode);
+        assertThatPolylinesAreEqual("y_`mEnmbbObA}@U]{@wAIMEOMHwBz@^p@Zn@Zx@Ld@DJLd@?@", leg.legGeometry.getPoints());
+
+        // if we reduce the stair penalty for wheelchairs we get a route that goes up the stairs
+        i = getTripPlan(start, end, r -> {
+            r.setMode(TraverseMode.WALK);
+            r.wheelchairStairsReluctance = 0;
+        }).itinerary.get(0);
+
+        leg = i.legs.get(0);
+        assertEquals("WALK", leg.mode);
+        assertThatPolylinesAreEqual("y_`mEnmbbO[XIHBFUTLd@?@", leg.legGeometry.getPoints());
+    }
+
+    @Test
+    public void canApplyNoWheelchairAccessStreetPenalty() {
+        // since i cannot find a real-life example of an inaccessible street in central Atlanta
+        // i manually set Hogue Street Northeast (https://www.openstreetmap.org/way/9254841)
+        // to be inaccessible in the OSM data
+
+        GenericLocation start = new GenericLocation(33.75545, -84.37105);
+        GenericLocation end = new GenericLocation(33.75767, -84.37120);
+
+        Itinerary i = getTripPlan(start, end, r -> r.setMode(TraverseMode.WALK)).itinerary.get(0);
+        Leg leg = i.legs.get(0);
+        assertEquals("WALK", leg.mode);
+        assertThatPolylinesAreEqual("sz_mE`v}aO?@?L?|AAzAM?A?G@SA[?I?E?C?E?oDCG?E?A?U?[?I??_@iAA@cAM?@o@?O", leg.legGeometry.getPoints());
+
+        // if we reduce the reduce the reluctance for wheelchair-inaccessible streets we get a route that uses
+        // Hogue Street Northeast
+        i = getTripPlan(start, end, r -> {
+            r.setMode(TraverseMode.WALK);
+            r.noWheelchairAccessOnStreetReluctance = 0;
+        }).itinerary.get(0);
+
+        leg = i.legs.get(0);
+        assertEquals("WALK", leg.mode);
+        assertEquals(0.5f, leg.accessibilityScore);
+        assertThatPolylinesAreEqual("sz_mE`v}aO?@M?cA?}A?uFCM??L?R", leg.legGeometry.getPoints());
```

Comment:
This test could use some assertions that checks for a correctly-calculated accessibility score derived from the way with wheelchair=no influencing the walk leg score.